airflow/www/views.py

```diff
@@ -1735,6 +1747,44 @@ def gantt(self):
             root=root,
         )
 
+    @expose('/redirect')
+    @login_required
+    @wwwutils.action_logging
+    def redirect(self):
+
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        redirect_to = request.args.get('redirect_to')
+        dttm = dateutil.parser.parse(execution_date)
+        dag = dagbag.get_dag(dag_id)
+
+        if not dag or task_id not in dag.task_ids:
+            flash(
+                "Task [{}.{}] doesn't seem to exist"
+                " at the moment".format(dag_id, task_id),
+                "error")
+            return redirect(request.referrer or '/admin/')
+
+        task = None
+
+        for t in dag.tasks:
+            if t.task_id == task_id:
+                task = t
+                break
+
+        url = task.get_redirect_url(task, dttm, redirect_to)
```

Comment:
This feels a bit odd to me, we invoke a function on the task, but we also pass the task itself to it. We already have this reference, right? In that case we can also get rid of the loop above, which is not that efficient (lineair instead of logarithmic).

Reply:
yeah, the `task` parameter seems obsolete, so I'll fix that. However, I don't find any other good way to get the `task` variable from the given information, so plz let me know if there is any.